pkg/conversation/scrub_streaming_test.go

```diff
@@ -0,0 +1,1153 @@
+				name:             "IP only",
+				input:            "Server at 192.168.1.100",
+				expectedScrubbed: []string{"192.168.1.100"},
+				expectedMissed:   []string{},
+			},
+			{
+				name:             "credit card only",
+				input:            "Card number 4263982640269299",
+				expectedScrubbed: []string{"4263982640269299"},
+				expectedMissed:   []string{},
+			},
+			{
+				name:             "IP + credit card together (IP first)",
+				input:            "IP is 192.168.1.100 and card 4263982640269299",
+				expectedScrubbed: []string{"192.168.1.100"},    // IP gets scrubbed
+				expectedMissed:   []string{"4263982640269299"}, // Card gets missed!
+			},
+			{
+				name:             "credit card + IP together (card first)",
+				input:            "Card 4263982640269299 on server 192.168.1.100",
+				expectedScrubbed: []string{}, // Need to determine what actually gets scrubbed
+				expectedMissed:   []string{}, // Need to determine what gets missed
+			},
+			{
+				name:             "phone + email (control test)",
+				input:            "Call +919140520809 or email test@example.com",
+				expectedScrubbed: []string{"+919140520809", "test@example.com"},
+				expectedMissed:   []string{},
+			},
+			{
+				name:             "phone + email + IP + card (complex scenario)",
+				input:            "Contact John at +919140520809 or email john@example.com. His IP is 192.168.1.100 and card 4263982640269299",
+				expectedScrubbed: []string{"+919140520809", "john@example.com", "192.168.1.100"}, // These work
+				expectedMissed:   []string{"4263982640269299"},                                   // Card still missed
+			},
+		}
+
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				scrubbed, err := scrubber.ScrubTexts([]string{tc.input})
+				require.NoError(t, err)
+				require.Len(t, scrubbed, 1)
+
+				result := scrubbed[0]
+				t.Logf("Input:  %s", tc.input)
+				t.Logf("Output: %s", result)
+
+				// For the "card first" case, let's discover the behavior
+				if tc.name == "credit card + IP together (card first)" {
+					cardMissed := strings.Contains(result, "4263982640269299")
+					ipMissed := strings.Contains(result, "192.168.1.100")
+
+					t.Logf("📊 DISCOVERY MODE:")
+					if cardMissed {
+						t.Logf("   Credit card: ❌ MISSED (4263982640269299 still visible)")
+					} else {
+						t.Logf("   Credit card: ✅ SCRUBBED")
+					}
+
+					if ipMissed {
+						t.Logf("   IP address:  ❌ MISSED (192.168.1.100 still visible)")
+					} else {
+						t.Logf("   IP address:  ✅ SCRUBBED")
+					}
+
+					if cardMissed && !ipMissed {
+						t.Logf("🔍 PATTERN: IP scrubbed, card missed (same as IP-first case)")
+					} else if !cardMissed && ipMissed {
+						t.Logf("🔍 PATTERN: Card scrubbed, IP missed (opposite of IP-first case)")
+					} else if cardMissed && ipMissed {
+						t.Logf("🔍 PATTERN: Both missed (mutual interference)")
+					} else {
+						t.Logf("🔍 PATTERN: Both scrubbed (order matters - this works!)")
+					}
```

Comment:
Is this unit test actually checking anything?

It seems clear that this file has been written with AI- but it doesn't look like the AI is actually asserting any test cases???